.github/workflows/ruby.yml

```diff
@@ -387,11 +392,28 @@ jobs:
         env:
           # We can invalidate the current cache by updating this.
           CACHE_VERSION: "2024-05-09"
+      - name: Setup NuGet credentials for vcpkg caching
+        shell: bash
+        run: |
+          `./vcpkg/vcpkg.exe fetch nuget | tail -n 1` \
+            sources add \
+            -source "https://nuget.pkg.github.com/$GITHUB_REPOSITORY_OWNER/index.json" \
+            -storepasswordincleartext \
+            -name "GitHub" \
+            -username "$GITHUB_REPOSITORY_OWNER" \
+            -password "${{ secrets.GITHUB_TOKEN }}"
+          `./vcpkg/vcpkg.exe fetch nuget | tail -n 1` \
+            setapikey "${{ secrets.GITHUB_TOKEN }}" \
+            -source "https://nuget.pkg.github.com/$GITHUB_REPOSITORY_OWNER/index.json"
+      - name: Build C++ VCPKG dependencies
+        shell: cmd
+        run: |
+          vcpkg\vcpkg.exe install --triplet x64-windows --x-manifest-root cpp --x-install-root build\cpp\vcpkg_installed
       - name: Build C++
         shell: cmd
         run: |
           call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64
-          bash -c "ci/scripts/cpp_build.sh $(pwd) $(pwd)/build"
+          bash -c "VCPKG_ROOT=\"$(pwd)/vcpkg\" ci/scripts/cpp_build.sh $(pwd) $(pwd)/build"
```

Comment:
Can we set this in `env:` something like `VCPKG_ROOT: "${{ github.workspace }}/vcpkg"`?
(Do we need UNIX style path here?)

Reply:
Yeah I think the reason I did it like this rather than in env was to use a unix path but I didn't test whether it works OK as a Windows path, I'll check.

Reply:
Yes it seems to work fine as a Windows path so I've set it in `env` now